pallets/subtensor/src/macros/dispatches.rs

```diff
@@ -2309,5 +2311,104 @@ mod dispatches {
             ensure_root(origin)?;
             Self::do_dissolve_network(netuid)
         }
+
+        /// --- Claims the root emissions for a coldkey.
+        /// # Args:
+        /// * 'origin': (<T as frame_system::Config>Origin):
+        /// 	- The signature of the caller's coldkey.
+        ///
+        /// # Event:
+        /// * RootClaimed;
+        /// 	- On the successfully claiming the root emissions for a coldkey.
+        ///
+        /// # Raises:
+        ///
+        #[pallet::call_index(121)]
+        #[pallet::weight((
+            Weight::from_parts(117_000_000, 7767)
+                .saturating_add(T::DbWeight::get().reads(12_u64))
+                .saturating_add(T::DbWeight::get().writes(4_u64)),
+            DispatchClass::Normal,
+            Pays::Yes
+        ))]
+        pub fn claim_root(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let coldkey: T::AccountId = ensure_signed(origin)?;
+
+            Self::maybe_add_coldkey_index(&coldkey);
+
+            let weight = Self::do_root_claim(coldkey);
+            Ok((Some(weight), Pays::Yes).into())
+        }
+
+        /// --- Sets the root claim type for the coldkey.
+        /// # Args:
+        /// * 'origin': (<T as frame_system::Config>Origin):
+        /// 	- The signature of the caller's coldkey.
+        ///
+        /// # Event:
+        /// * RootClaimTypeSet;
+        /// 	- On the successfully setting the root claim type for the coldkey.
+        ///
+        #[pallet::call_index(122)]
+        #[pallet::weight((
+            Weight::from_parts(19_420_000, 0).saturating_add(T::DbWeight::get().writes(4_u64)),
+            DispatchClass::Normal,
+            Pays::Yes
+        ))]
+        pub fn set_root_claim_type(
+            origin: OriginFor<T>,
+            new_root_claim_type: RootClaimTypeEnum,
+        ) -> DispatchResult {
+            let coldkey: T::AccountId = ensure_signed(origin)?;
+
+            Self::maybe_add_coldkey_index(&coldkey);
+
+            Self::change_root_claim_type(&coldkey, new_root_claim_type);
+            Ok(())
+        }
+
+        /// --- Sets root claim number (sudo extrinsic).
+        #[pallet::call_index(123)]
+        #[pallet::weight((
+            Weight::from_parts(4_000_000, 0).saturating_add(T::DbWeight::get().writes(1_u64)),
+            DispatchClass::Operational,
+            Pays::Yes
+        ))]
+        pub fn sudo_set_num_root_claims(origin: OriginFor<T>, new_value: u64) -> DispatchResult {
+            ensure_root(origin)?;
+
+            ensure!(
+                new_value > 0 && new_value <= MAX_NUM_ROOT_CLAIMS,
+                Error::<T>::InvalidNumRootClaim
+            );
```

Comment:
```suggestion
                new_value <= MAX_NUM_ROOT_CLAIMS,
                Error::<T>::InvalidNumRootClaim
            );
```

We need to allow the option to set NumRootClaim=0, not 1 (as a minimum).
It's very difficult and time-consuming to create 100-1000 "stakers" just to TRY to prevent a single Cold Key from ending up in the random auto claim pool. But even a large number of delegate keys doesn't guarantee a successful test - flaky behavior will still occur.
Allowing the option to set 0 guarantees the ability to test two behaviors: with and without a random auto-claim pool.

Or you could implement some function/constant for this purpose like this
```rust
pub const MIN_NUM_ROOT_CLAIMS: u64 = prod_or_fast!(1, 0);
```

Otherwise, we won't be able to implement 100% stable tests.

Reply:
I implemented "zero as acceptable value that disables auto-claim" option.